econml/iv/dr/_dr.py

```diff
@@ -73,83 +246,1246 @@ def _effect_estimate(self, nuisances):
+        Parameters
+        ----------
+        Y: (n,) vector of length n
+            Outcomes for each sample
+        T: (n,) vector of length n
+            Treatments for each sample
+        Z: (n, d_z) matrix
+            Instruments for each sample
+        X: optional(n, d_x) matrix or None (Default=None)
+            Features for each sample
+        W: optional(n, d_w) matrix or None (Default=None)
+            Controls for each sample
+        sample_weight : (n,) array like, default None
+            Individual weights for each sample. If None, it assumes equal weight.
+        groups: (n,) vector, optional
+            All rows corresponding to the same group will be kept together during splitting.
+            If groups is not None, the `cv` argument passed to this class's initializer
+            must support a 'groups' argument to its split method.
+        cache_values: bool, default False
+            Whether to cache inputs and first stage results, which will allow refitting a different final model
+        inference: string, :class:`.Inference` instance, or None
+            Method for performing inference.  This estimator supports ``'bootstrap'``
+            (or an instance of :class:`.BootstrapInference`) and ``'debiasedlasso'``
+            (or an instance of :class:`.LinearModelInferenceDiscrete`).
 
-        theta_dr, clipped_cov = self._effect_estimate(nuisances)
+        Returns
+        -------
+        self
+        """
+        # TODO: support freq_weight and sample_var in debiased lasso
+        # Replacing fit from _OrthoLearner, to reorder arguments and improve the docstring
+        check_high_dimensional(X, T, threshold=5, featurizer=self.featurizer,
+                               discrete_treatment=self.discrete_treatment,
+                               msg="The number of features in the final model (< 5) is too small for a sparse model. "
+                               "We recommend using the LinearDRLearner for this low-dimensional setting.")
+        return super().fit(Y, T, X=X, W=W, Z=Z,
+                           sample_weight=sample_weight, groups=groups,
+                           cache_values=cache_values, inference=inference)
 
-        if (X is not None) and (self._featurizer is not None):
-            X = self._featurizer.fit_transform(X)
-        if self._opt_reweighted and (sample_weight is not None):
-            sample_weight = sample_weight * clipped_cov.ravel()**2
-        elif self._opt_reweighted:
-            sample_weight = clipped_cov.ravel()**2
-        self._model_final.fit(X, theta_dr, **filter_none_kwargs(sample_weight=sample_weight,
-                                                                freq_weight=freq_weight, sample_var=sample_var))
+    @property
+    def fit_cate_intercept_(self):
+        return self.ortho_learner_model_final_._fit_cate_intercept
 
-        return self
+    @property
```

Comment:
If we switch this should be removed and the parent implementation will always return False.